apps/desktop/src/components/ReduxResult.svelte

```diff
@@ -1,19 +1,27 @@
-<script lang="ts">
+<script lang="ts" generics="A">
 	import Icon from '@gitbutler/ui/Icon.svelte';
 	import type { QueryStatus } from '@reduxjs/toolkit/query';
 	import type { Snippet } from 'svelte';
 
-	type Props = {
+	type Props<A> = {
 		status: QueryStatus;
-		error: unknown;
-		children: Snippet;
+		error?: unknown;
+		data?: A;
+		children: Snippet<[A]>;
+		empty: Snippet;
 	};
```

Comment:
I'm wondering about this API. Is it going to get tedious always restructuring before passing to your result?

I'm just imagining if I had two or three queries in one component, I'd much rather say
```
const foo = getFoo()
const bar = getBar()
const baz = getBaz()
```

rather than

```
const { data: fooData, error: fooError, status: fooStatus } = getFoo()
const { data: barData, error: barError, status: barStatus } = getBar()
const { data: bazData, error: bazError, status: bazStatus } = getBaz()
```

If we work with the whole response object, it also means we can define functions like `map` and `combine` like we have in the web project